model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/entities/UserSessionEntity.java

```diff
@@ -104,6 +105,14 @@ public void setIpAddress(String ipAddress) {
         this.ipAddress = ipAddress;
     }
 
+    public DeviceInfo getDeviceInfo() {
+        return new DeviceInfo(this.getNotes().get(DeviceInfo.ID));
```

Comment:
Is it possible to add check for "nullity" here? And either return null or empty instance to the caller in that case? So something like:
````
String deviceInfoStr = this.getNotes().get(DeviceInfo.ID);
return deviceInfoStr == null ? null : new DeviceInfo(deviceInfoStr);
````
or second line to: 
````
return deviceInfoStr == null ? DeviceInfo.EMPTY_INSTANCE : new DeviceInfo(deviceInfoStr);
````
The only reason for this is serialization when reading UserSessionEntity from the previous version (may happen in some scenarios like cross-dc).